docs/upgrade.html

```diff
@@ -47,14 +47,16 @@
 
 <p><b>Note:</b> Bumping the protocol version and restarting can be done any time after the brokers were upgraded. It does not have to be immediately after.
 
-<ol>
-    <li>Upgrading a Kafka Streams Applications:
-        <ul>
-            <li>You need to recompile your code. Just swapping the jar file will not work and will break your appliation.</li>
-            <li>If you use a custom timestamp extractor, you will need to update this code, because the <code>TimestampExtractor</code> interface was changed.</li>
-        </ul>
-    </li>
-</ol>
+<h5><a id="upgrade_1020_streams" href="#upgrade_1020_streams">Upgrading a Kafka Streams Application</a></h5>
+<ul>
+    <li> You need to recompile your code. Just swapping the jar file will not work and will break your application. </li>
```

Comment:
Also, the other document (streams.html) does not include the information that you must recompile your code etc.  I think, as Damian suggestioned, we should cover the upgrade info comprehensively in one place, and then link to it (perhaps we could highlight 2-3 key changes before showing the link "to read more")